x-pack/plugins/security_solution/public/common/hooks/use_invalid_filter_query.tsx

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { useEffect } from 'react';
+import { Query } from 'src/plugins/data/public';
+import { useAppToasts } from './use_app_toasts';
+
+/**
+ * Adds a toast error message whenever invalid KQL is submitted through the search bar
+ */
+export const useInvalidFilterQuery = ({
+  filterQuery,
+  kqlError,
+  query,
+  startDate,
+  endDate,
+}: {
+  filterQuery?: string;
+  kqlError?: Error;
+  query: Query;
+  startDate: string;
+  endDate: string;
+}) => {
+  const { addError } = useAppToasts();
+
+  useEffect(() => {
+    if (filterQuery === undefined && kqlError != null) {
+      // Removes error stack from user view
+      delete kqlError.stack;
```

Comment:
Ideally we might avoid mutating the `kqlError` arg, because consumers of the hook are probably not expecting this side effect.

As an alternative to using `delete` to remove the `stack`, I would typically recommend using a utility function like `omit` from `lodash/fp`, because it's guaranteed not to mutate the original, and it's less verbose than the native TS alternative. Unfortunately when I tried that, the error toaster didn't behave as expected, because the implementation of `addError` uses `IEsError` in a TS type guard.

I experimented with a few different options for safely cloning and/or constructing a new error (that omits the `stack`), but each path I explored had some unexpected "gotchas". 

To keep this PR moving forward, consider documenting the side effect of mutating `kqlError` if we can't reasonably eliminate it.